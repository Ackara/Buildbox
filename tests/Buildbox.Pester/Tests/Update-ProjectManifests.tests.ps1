Import-Module "$(Split-Path $PSScriptRoot -Parent)\config.psm1" -Force
$context = New-TestEnvironment -useTemp;
Import-Module $context.ModulePath -Force;

Describe "Update-ProjectManifests" {
	Push-Location $context.TestDir;

	&git init;
	&git add *;
	&git commit -m init;
	
	$manifestPath = "$($context.TestDataDir)\manifest-full.json";
	$result = Get-Item $manifestPath | Get-BuildboxManifest | Update-ProjectManifests $context.TestDir -Commit -Tag -Minor;

	Context "Update Project" {
		It "should increment manifest version number" {
			$result.Manifest.Version.ToString() | Should Be "1.3.0";
			{ Approve-File $manifestPath } | Should Not Throw;
		}

		It "should update a .net core project." {
			$csprojFile = Get-Item "$($context.TestDataDir)\*standard*.csproj";
			{ Approve-File $csprojFile.FullName } | Should Not Throw;
		}

		It "should update a .net 4+ project" {
			$csprojFile = Get-Item "$($context.TestDataDir)\*\*\*Info.cs";
			{ Approve-File $csprojFile.FullName } | Should Not Throw;
		}

		It "should update a vsix project manifest" {
			$csprojFile = Get-Item "$($context.TestDataDir)\*.vsixmanifest";
			{ Approve-File $csprojFile.FullName } | Should Not Throw;
		}

		It "should update a powershell module manifest" {
			$psManifest = Get-Item "$($context.TestDataDir)\*.psd1";
			{ Approve-File $psManifest.FullName } | Should Not Throw;
		}
	}

	Context "Git" {
		It "should return a list of the modified files." {
			$result.ModifiedFiles.Count | Should BeGreaterThan 2;
		}

		It "should commit modified files to source control" {
			$stauts = &git status | Out-String;
			$stauts | Should Match "nothing to commit";
		}

		It "should tag commit with version number" {
			$tag = &git tag | Out-String;
			$tag | Should Match "v$($result.Manifest.Version.ToString())";
		}
	}
	Pop-Location;
}