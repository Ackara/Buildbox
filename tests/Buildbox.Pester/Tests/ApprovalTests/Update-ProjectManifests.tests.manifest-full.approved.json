{
    "id":  "Acklann.Buildbox",
    "productName":  "Buildbox",
    "description":  "This is a informative description.",
    "authors":  "Ackara",
    "owner":  "Ackara",
    "copyright":  "All rights reserved.",
    "projectUrl":  "https://example.com/project/buildbox",
    "iconUri":  "https://example.com/icon/buildbox.png",
    "licenseUri":  "https://example.com/buildbox/license.txt",
    "tags":  "build, deploy",
    "releaseNotes":  "some notes",
    "version":  {
                    "Major":  1,
                    "Minor":  3,
                    "Patch":  0
                },
    "branchSuffixMap":  {
                            "*":  "alpha",
                            "dev":  "preview",
                            "master":  ""
                        }
}
